File: reorder.js
const fs = require('fs');

const path = 'index.html';
let content = fs.readFileSync(path, 'utf8');

const coverOld = '<div class="s-sub ac-reveal" style="--i:3;line-height:1.8">5.1 - Phòng Hàng không Không gian<br>5.2 - Phòng Thương mại điện tử<br>5.3 - Phòng Phát triển kinh doanh</div>';
const coverNew = '<div class="s-sub ac-reveal" style="--i:3;line-height:1.8">5.1 - Phòng Phát triển kinh doanh<br>5.2 - Phòng Hàng không Không gian<br>5.3 - Phòng Thương mại điện tử</div>';

content = content.replace(coverOld, coverNew);

const startMarker = '<!-- ===== S14a Khó khăn HKKG (1) ===== -->';
const midMarker1 = '<!-- ===== S16 VAI TRÒ TMĐT ===== -->';
const midMarker2 = '<!-- ===== S17c Khó khăn PTKD (5.3) ===== -->';
const endMarker = '<!-- ===== S18 THANK YOU ===== -->';

const idxStart = content.indexOf(startMarker);
const idxMid1 = content.indexOf(midMarker1);
const idxMid2 = content.indexOf(midMarker2);
const idxEnd = content.indexOf(endMarker);

if (idxStart === -1 || idxMid1 === -1 || idxMid2 === -1 || idxEnd === -1) {
    console.error("Could not find markers");
    process.exit(1);
}

let hkkgText = content.substring(idxStart, idxMid1);
let tmdtText = content.substring(idxMid1, idxMid2);
let ptkdText = content.substring(idxMid2, idxEnd);

hkkgText = hkkgText.replace(/5\.1 · Phòng Hàng không Không gian/g, '5.2 · Phòng Hàng không Không gian');
ptkdText = ptkdText.replace(/5\.3 · Phòng Phát triển kinh doanh/g, '5.1 · Phòng Phát triển kinh doanh');

const newBlocks = ptkdText + hkkgText + tmdtText;

const newContent = content.substring(0, idxStart) + newBlocks + content.substring(idxEnd);

fs.writeFileSync(path, newContent, 'utf8');
console.log("Success");

File: reorder.py
import re

path = r'c:\Users\vietdt\OneDrive - PetroVietnam Exploration Production Corporation (PVEP)\Desktop\HTML\PVI-main\index.html'
with open(path, 'r', encoding='utf-8') as f:
    content = f.read()

cover_old = r'<div class="s-sub ac-reveal" style="--i:3;line-height:1.8">5\.1 - Phòng Hàng không Không gian<br>5\.2 - Phòng Thương mại điện tử<br>5\.3 - Phòng Phát triển kinh doanh</div>'
cover_new = r'<div class="s-sub ac-reveal" style="--i:3;line-height:1.8">5.1 - Phòng Phát triển kinh doanh<br>5.2 - Phòng Hàng không Không gian<br>5.3 - Phòng Thương mại điện tử</div>'
content = re.sub(cover_old, cover_new, content)

hkkg_match = re.search(r'(<!-- ===== S14a Khó khăn HKKG \(1\) ===== -->.*?)(?=<!-- ===== S16 VAI TRÒ TMĐT ===== -->)', content, re.DOTALL)
tmdt_match = re.search(r'(<!-- ===== S16 VAI TRÒ TMĐT ===== -->.*?)(?=<!-- ===== S17c Khó khăn PTKD \(5\.3\) ===== -->)', content, re.DOTALL)
ptkd_match = re.search(r'(<!-- ===== S17c Khó khăn PTKD \(5\.3\) ===== -->.*?)(?=<!-- ===== S18 THANK YOU ===== -->)', content, re.DOTALL)

if not hkkg_match or not tmdt_match or not ptkd_match:
    print('Could not find blocks')
    if not hkkg_match: print('No HKKG')
    if not tmdt_match: print('No TMĐT')
    if not ptkd_match: print('No PTKD')
else:
    hkkg_text = hkkg_match.group(1)
    tmdt_text = tmdt_match.group(1)
    ptkd_text = ptkd_match.group(1)
    
    hkkg_text = hkkg_text.replace('5.1 · Phòng Hàng không Không gian', '5.2 · Phòng Hàng không Không gian')
    ptkd_text = ptkd_text.replace('5.3 · Phòng Phát triển kinh doanh', '5.1 · Phòng Phát triển kinh doanh')
    
    new_blocks = ptkd_text + hkkg_text + tmdt_text
    
    start_idx = hkkg_match.start()
    end_idx = ptkd_match.end()
    
    new_content = content[:start_idx] + new_blocks + content[end_idx:]
    
    with open(path, 'w', encoding='utf-8') as f:
        f.write(new_content)
    print('Success')
